test/exercism/user_contribution_test.rb

```diff
@@ -0,0 +1,37 @@
+require_relative '../test_helper'
+require_relative '../x_helper'
+require_relative '../integration_helper'
+
+class UserContributionTest < Minitest::Test
+  include DBCleaner
+
+  def setup
+    super
+    @user = User.create
+    @f = './test/fixtures/xapi_v3_tracks.json'
+    create_exercise('animal', 'dog')
+    create_exercise('fake', 'hello-world')
+    create_exercise('fake', 'two')
+  end
+
+  def test_user_contributions
+    X::Xapi.stub(:get, [200, File.read(@f)]) do
+      actual = UserContribution.new(@user).list
+      expected = {
+        completed_tracks: [
```

Comment:
It seems like if we're only listing out which tracks are completed, we could filter the track objects instead of creating an entirely new data structure. What data does the view actually need? The track id and language, right?